src/errors/groutError.ts

```diff
@@ -0,0 +1,14 @@
+import { ErrorType, ErrorTypeStatuses } from "./errorType";
+
+export class GroutError extends Error {
+    status: number;
+    errorType: ErrorType;
```

Comment:
Do we need to store both of these? As Mantra said they have a 1-1 mapping?

Or is it that the status can be any valid HTTP status, not just a defined error type status?

Reply:
Maybe e.g. status should be a getter for the error type, or vice versa. (rather than a property assigned by the constructor)

Reply:
I do think they're a bit different conceptually, and I think it's ok for GroutError to manage that. There may come a time when it's not a 1-1 mapping, though it is at the moment. OK, will change to getter as that will cover both cases.